tardis/transport/montecarlo/tests/test_trackers.py

```diff
@@ -0,0 +1,144 @@
+
+
+def test_tracking_from_rpacket_tracker_simulation(
+    simulation_rpacket_tracking_enabled,
+):
+    sim = simulation_rpacket_tracking_enabled
+    rpacket_trackers = sim.transport.transport_state.rpacket_tracker
+    rpacket_last_interaction_trackers = (
+        sim.transport.transport_state.rpacket_last_interaction_trackers
+    )
+    no_of_packets = len(rpacket_trackers)
+
+    index_from_rpacket_tracker = np.empty(no_of_packets, dtype=np.int64)
+    index_from_last_interaction_tracker = np.empty(
+        no_of_packets, dtype=np.int64
+    )
+
+    r_from_rpacket_tracker = np.empty(no_of_packets, dtype=np.float64)
+    r_from_last_interaction_tracker = np.empty(no_of_packets, dtype=np.float64)
+
+    nu_from_rpacket_tracker = np.empty(no_of_packets, dtype=np.float64)
+    nu_from_last_interaction_tracker = np.empty(no_of_packets, dtype=np.float64)
+
+    energy_from_rpacket_tracker = np.empty(no_of_packets, dtype=np.float64)
+    energy_from_last_interaction_tracker = np.empty(
+        no_of_packets, dtype=np.float64
+    )
+
+    shell_id_from_rpacket_tracker = np.empty(no_of_packets, dtype=np.int64)
+    shell_id_from_last_interaction_tracker = np.empty(
+        no_of_packets, dtype=np.int64
+    )
+
+    type_from_rpacket_tracker = np.empty(no_of_packets, dtype=np.int64)
+    type_from_last_interaction_tracker = np.empty(no_of_packets, dtype=np.int64)
+
+    for i in range(no_of_packets):
+        interactions = rpacket_trackers[i].num_interactions
+
+        index_from_rpacket_tracker[i] = rpacket_trackers[i].index
+        index_from_last_interaction_tracker[
+            i
+        ] = rpacket_last_interaction_trackers[i].index
+
+        r_from_rpacket_tracker[i] = rpacket_trackers[i].r[interactions - 1]
+        r_from_last_interaction_tracker[i] = rpacket_last_interaction_trackers[
+            i
+        ].r
+
+        nu_from_rpacket_tracker[i] = rpacket_trackers[i].nu[interactions - 1]
+        nu_from_last_interaction_tracker[i] = rpacket_last_interaction_trackers[
+            i
+        ].nu
+
+        energy_from_rpacket_tracker[i] = rpacket_trackers[i].energy[
+            interactions - 1
+        ]
+        energy_from_last_interaction_tracker[
+            i
+        ] = rpacket_last_interaction_trackers[i].energy
+
+        shell_id_from_rpacket_tracker[i] = rpacket_trackers[i].shell_id[
+            interactions - 1
+        ]
+        shell_id_from_last_interaction_tracker[
+            i
+        ] = rpacket_last_interaction_trackers[i].shell_id
+
+        type_from_rpacket_tracker[i] = rpacket_trackers[i].interaction_type[
+            interactions - 1
+        ]
+        type_from_last_interaction_tracker[
+            i
+        ] = rpacket_last_interaction_trackers[i].interaction_type
+
+    npt.assert_array_equal(
+        index_from_rpacket_tracker, index_from_last_interaction_tracker
+    )
+    npt.assert_allclose(
+        r_from_rpacket_tracker, r_from_last_interaction_tracker, rtol=1e-14
```

Comment:
why `rtol`

Reply:
Used the default values provided by numpy.testing